app/src/androidTest/java/co/nimblehq/compose/crypto/ui/screen/detail/DetailScreenUiTest.kt

```diff
@@ -0,0 +1,146 @@
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import kotlin.math.abs
+
+@ExperimentalCoroutinesApi
+class DetailScreenUiTest {
+
+    @get:Rule
+    val composeAndroidTestRule = createAndroidComposeRule<MainActivity>()
+
+    private val errorGeneric: String
+        get() = composeAndroidTestRule.activity.getString(R.string.error_generic)
+
+    private val mockGetCoinDetailUseCase: GetCoinDetailUseCase = mockk()
+    private val mockGetCoinPricesUseCase: GetCoinPricesUseCase = mockk()
+
+    private lateinit var detailViewModel: DetailViewModel
+
+
+    @Before
+    fun setUp() {
+        every { mockGetCoinDetailUseCase.execute(any()) } returns flowOf(coinDetail)
+        every { mockGetCoinPricesUseCase.execute(any()) } returns flowOf(coinPrices)
+
+        initViewModel()
+        composeAndroidTestRule.activity.setContent {
+            DetailScreen(
+                coinId = "",
+                viewModel = detailViewModel,
+                navigator = {}
+            )
+        }
+    }
+
+    @Test
+    fun when_navigating_to_detail_screen_it_shows_loading() {
+        every { mockGetCoinDetailUseCase.execute(any()) } returns flow {
+            delay(100L)
+            emit(coinDetail)
+        }
+        every { mockGetCoinPricesUseCase.execute(any()) } returns flow {
+            delay(100L)
+            emit(coinPrices)
+        }
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(TestTagDetailLoading).assertIsDisplayed()
+        }
+    }
+
+    @Test
+    fun when_navigating_to_detail_screen_it_renders_chart_interval_buttons_properly() {
+        with(composeAndroidTestRule) {
+            onNodeWithText(TimeIntervals.ONE_DAY.text).assertIsDisplayed()
+            onNodeWithText(TimeIntervals.ONE_WEEK.text).assertIsDisplayed()
+            onNodeWithText(TimeIntervals.ONE_MONTH.text).assertIsDisplayed()
+            onNodeWithText(TimeIntervals.ONE_YEAR.text).assertIsDisplayed()
+            onNodeWithText(TimeIntervals.FIVE_YEAR.text).assertIsDisplayed()
+        }
+    }
+
+    @Test
+    fun when_navigating_to_detail_screen_it_render_currentPrice_and_priceChangePercentage24hInCurrency_properly() {
+        with(composeAndroidTestRule) {
+            coinDetail.marketData?.let { marketData ->
+                val currentPrice = "$${marketData.currentPrice[FIAT_CURRENCY]?.toFormattedString()}"
+                onNodeWithText(currentPrice).assertIsDisplayed()
+
+                val priceChangePercentage24hInCurrency = this.activity.getString(
+                    R.string.coin_profit_percent,
+                    abs(marketData.priceChangePercentage24hInCurrency[FIAT_CURRENCY] ?: 0.0).toFormattedString()
+                )
+                onNodeWithText(priceChangePercentage24hInCurrency).assertIsDisplayed()
+            }
+        }
```

Comment:
Let's convert to `UIModel` So, we don't need to handle null 🙏 
```suggestion
val coinUiModel = coinDetail.toUiModel()
                val currentPrice = "$${coinUiModel.currentPrice.toFormattedString()}"
                onNodeWithText(currentPrice).assertIsDisplayed()

                val priceChangePercentage24hInCurrency = "${abs(coinUiModel.priceChangePercentage24hInCurrency).toFormattedString()}%"
                onNodeWithText(priceChangePercentage24hInCurrency).assertIsDisplayed()
            }
```